timesketch/api/v1/resources.py

```diff
@@ -1753,8 +1782,51 @@ def post(self):
             searchindex, status_code=HTTP_STATUS_CODE_CREATED)
 
 
+class AnalysisResource(ResourceMixin, Resource):
+    """Resource to get analyzer session."""
+
+    @login_required
+    def get(self, sketch_id, timeline_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            An analysis in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        timeline = Timeline.query.get(timeline_id)
+        analyses = Analysis.query.filter_by(timeline=timeline).all()
+
+        if not sketch.has_permission(current_user, 'read'):
```

Comment:
can you move this above .... you've got

sketch = ....
if not sketch....

that is to abort early, before you start doing any queries, if you know the user doesn't even have the aCL to perform the action

Reply:
Done